### ISSUE TITLE
v0.6.0 doesn't compile from crates.io due to libgit2-sys dep

### ISSUE THREAD

**Issue by kbknapp** [rust-lang/cargo#2162]
`libgit2-sys` recently updated to v0.3.7 which had a "breaking change" for `cargo` v0.6.0

Trying to compile results in:

```
[...]/cargo-0.6.0/src/cargo/sources/path.rs:147:17: 147:65 error: unresolved import `libgit2_sys::git_filemode_t::GIT_FILEMODE_COMMIT`. 

There is no `GIT_FILEMODE_COMMIT` in `libgit2_sys::git_filemode_t` [E0432]

[...]/cargo-0.6.0/src/cargo/sources/path.rs:147             

use libgit2_sys::git_filemode_t::GIT_FILEMODE_COMMIT;
                                                                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[...]/cargo-0.6.0/src/cargo/sources/path.rs:147:17: 147:65 help: run `rustc --explain E0432` to see a detailed explanation
error: aborting due to previous error
```

Problem arose because my `Cargo.lock` for a project specifies v0.3.6 of `libgit2-sys` as a child dep of `cargo` but when performing `cargo package` I'm assuming it ignores `Cargo.lock` files becasue `libgit2-sys` gets bumped to v0.3.7 which causes this error.

**Comment by alexcrichton** [comment]
I just published 0.6.1 which should fix this, thanks for the report!

**Comment by kbknapp** [comment]
:+1: thanks!
